ion-hash/tests/ion_hash_tests.rs

```diff
@@ -1,71 +1,96 @@
 // Copyright Amazon.com, Inc. or its affiliates.
 
-use ion_hash::{self, IonHashDigest, IonHasher};
+use digest::{consts::U256, generic_array::GenericArray, Digest, Output};
+use ion_hash::{self, IonHasher};
 use ion_rs::result::IonResult;
 use ion_rs::value::loader::{loader, Loader};
 use ion_rs::value::*;
-use std::{
-    fs::read,
-    sync::{Arc, Mutex},
-};
+use std::fs::read;
 
 #[derive(Default, Clone)]
 struct TestDigest {
-    inner: Arc<Mutex<TestDigestInner>>,
+    updates: GenericArray<u8, U256>,
+    position: usize,
 }
 
-#[derive(Default)]
-struct TestDigestInner {
-    updates: Vec<u8>,
-    digest: Vec<u8>,
-}
+impl Digest for TestDigest {
+    // Pick a number bigger than the biggest test case digest expectation.
+    type OutputSize = U256;
 
-impl TestDigest {
-    fn inspect(&self) -> (Vec<u8>, Vec<u8>) {
-        let inner = self.inner.lock().unwrap();
-        (inner.updates.clone(), inner.digest.clone())
+    fn new() -> Self {
+        Self {
+            updates: GenericArray::default(),
+            position: 0,
+        }
     }
-}
 
-impl IonHashDigest for TestDigest {
     fn update(&mut self, bytes: impl AsRef<[u8]>) {
-        let mut inner = self.inner.lock().unwrap();
         for b in bytes.as_ref() {
-            inner.updates.push(*b);
-            inner.digest.push(*b);
+            self.updates[self.position] = *b;
+            self.position += 1;
```

Comment:
This really just looks like `Vec::push` to me...

Reply:
It is. See the other comment. We either work with a generic array internally and just return it; or we work with a vec internally and then convert it over to a generic array later. I think the above is actually simpler.